main.go

```diff
@@ -225,6 +227,23 @@ func main() {
 			setupLog.Error(err, "unable to create webhook", "webhook", "KubeadmControlPlane")
 			os.Exit(1)
 		}
+
+		if err = (&kubeadmbootstrapv1alpha3.KubeadmConfig{}).SetupWebhookWithManager(mgr); err != nil {
+			setupLog.Error(err, "unable to create webhook", "webhook", "KubeadmConfig")
+			os.Exit(1)
+		}
+		if err = (&kubeadmbootstrapv1alpha3.KubeadmConfigList{}).SetupWebhookWithManager(mgr); err != nil {
+			setupLog.Error(err, "unable to create webhook", "webhook", "KubeadmConfigList")
+			os.Exit(1)
+		}
+		if err = (&kubeadmbootstrapv1alpha3.KubeadmConfigTemplate{}).SetupWebhookWithManager(mgr); err != nil {
+			setupLog.Error(err, "unable to create webhook", "webhook", "KubeadmConfigTemplate")
+			os.Exit(1)
+		}
+		if err = (&kubeadmbootstrapv1alpha3.KubeadmConfigTemplateList{}).SetupWebhookWithManager(mgr); err != nil {
```

Comment:
It looks like for the other conversion webhooks we are setting up the webhook through the v1alpha2 types, but the approach here feels more correct to me (since v1alpha3 is being used as the hub type). I'm wondering if we need to change the other conversion webhooks to be configured through the v1alpha3 types as well.

Reply:
IMO it would make more sense to have all the conversion webhooks configured through the same hub type version (_v1alpha3_ in this case).